modules/processing.py

```
@@ -0,0 +1,28 @@
+
+def auth_required(fn):
+    def wrapper(*args, **kwargs):
+        if not get_current_user():
+            raise Error("Authentication required")
+        return fn(*args, **kwargs)
+    return wrapper
+
+@auth_required
+def save_image(image, metadata):
+    output_dir = get_output_directory()
+    filename = generate_unique_filename()
```

Ruff (0.8.2)

Undefined name `generate_unique_filename`

(F821)